cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/AnnotateDiagnostics.java

```diff
@@ -0,0 +1,180 @@
+
+    public static String renderDiagnostic(Diagnostic diagnostic, Document document, int terminalWidth) {
+
+        String diagnosticCode = diagnostic.diagnosticInfo().code();
+        if (diagnosticCode.startsWith("BCE")) {
+            int diagnosticCodeNumber = Integer.parseInt(diagnosticCode.substring(3));
+            if (diagnosticCodeNumber < 1000) {
+                PackageDiagnostic packageDiagnostic = (PackageDiagnostic) diagnostic;
+                return diagnosticToString(diagnostic) + "\n" + getSyntaxDiagnosticAnnotation(
+                        document, packageDiagnostic, diagnosticCodeNumber, terminalWidth);
+            }
+        }
+        DiagnosticAnnotation diagnosticAnnotation = getDiagnosticLineFromSyntaxAPI(
+                document, diagnostic.location(), diagnostic.diagnosticInfo().severity(), terminalWidth);
+        return diagnosticToString(diagnostic) + "\n" + diagnosticAnnotation;
+
+    }
+
+    public static String renderDiagnostic(Diagnostic diagnostic) {
+        return diagnosticToString(diagnostic);
+    }
+
+    private static String diagnosticToString(Diagnostic diagnostic) {
+        DiagnosticSeverity severity = diagnostic.diagnosticInfo().severity();
+        String severityString = severity.toString();
+        String color = SEVERITY_COLORS.get(severity);
+        String message = diagnostic.toString().substring(severityString.length());
+        String code = diagnostic.diagnosticInfo().code();
+
+        return String.format("@|%s %s|@%s (%s)", color, severityString, message, code);
+    }
+
+    private static DiagnosticAnnotation getDiagnosticLineFromSyntaxAPI(Document document, Location location,
+                                                                       DiagnosticSeverity severity, int terminalWidth) {
+        TextDocument textDocument = document.textDocument();
+        int startOffset = location.lineRange().startLine().offset();
+        int endOffset = location.lineRange().endLine().offset();
+        int startLine = location.lineRange().startLine().line();
+        int endLine = location.lineRange().endLine().line();
+
+        if (startLine != endLine) {
+            return new DiagnosticAnnotation(
+                    getLines(textDocument, startLine, endLine),
+                    startOffset,
+                    textDocument.line(startLine).length() - startOffset,
+                    endOffset,
+                    startLine + 1,
+                    severity,
+                    DiagnosticAnnotation.DiagnosticAnnotationType.REGULAR,
+                    terminalWidth);
+        }
+        int length = endOffset - startOffset;
+        return new DiagnosticAnnotation(
+                getLines(textDocument, startLine, endLine),
+                startOffset,
+                length == 0 ? 1 : length,
+                startLine + 1,
+                severity,
+                DiagnosticAnnotation.DiagnosticAnnotationType.REGULAR,
+                terminalWidth);
```

Comment:
```suggestion
        int length = startLine != endLine ? 
                            textDocument.line(startLine).length() - startOffset 
                            : endOffset - startOffset;
        
        return new DiagnosticAnnotation(
                getLines(textDocument, startLine, endLine),
                startOffset,
                length == 0 ? 1 : length,
                startLine + 1,
                severity,
                DiagnosticAnnotation.DiagnosticAnnotationType.REGULAR,
                terminalWidth);
```

this type of logic can reduce the codelines